apps/pie-storybook/utilities/index.ts

```diff
@@ -105,34 +105,42 @@ export const createVariantStory = <T extends Record<string, any>>(
 
             const backgroundValue = CUSTOM_BACKGROUNDS.values.find((bg: BackgroundValue) => bg.name === storyOpts?.bgColor)?.value || '#ffffff';
 
+            const isMultiColumn = storyOpts?.multiColumn || false;
+
             return html`
-        <div style="display: block; width: 100%;">
-            ${propCombinations.map((props) => {
+            <div style="display: flex; flex-wrap: wrap; width: 100%;">
+                ${propCombinations.map((props) => {
                 const typedProps = props as T;
 
                 return html`
-                    <div style="border: 1px solid black; padding: 16px; margin-bottom: 16px; width: 100%; box-sizing: border-box;">
-                        <div style="display: grid; grid-template-columns: repeat(2, 1fr); gap: 8px; font-family: monospace; background-color: #f9f9f9; padding: 8px; border-radius: 4px;">
-                            ${Object.entries(typedProps).map(([key, value]) => html`
-                                <div><strong>${key}:</strong> ${JSON.stringify(value)}</div>
-                            `)}
-                        </div>
-                        <div
-                          style="
-                            margin-top: 16px;
-                            border: 2px dashed #aaa;
-                            padding: 8px;
-                            border-radius: 4px;
-                            background-color: ${backgroundValue};
-                          "
-                        >
-                            ${template({ ...typedProps })}
+                        <div style="
+                            border: 1px solid black;
+                            padding: 16px;
+                            margin: 8px;
+                            box-sizing: border-box;
+                            ${isMultiColumn ? 'flex: 1 1 calc(33.33% - 16px); height: 300px; max-width: calc(33.33% - 16px);' : 'width: 100%;'}
```

Comment:
you could probably force a 3-column grid with CSS grid a bit easier, it should avoid having to use these calculations!